content/v1.17/guides/extensions-release-process.md

```diff
@@ -0,0 +1,352 @@
+which overrides inferring from a git ref or tag. The [`ci.yml`](https://github.com/crossplane-contrib/function-python/blob/main/.github/workflows/ci.yml)
+file for `crossplane-contrib/function-python` is a good example.
+{{< /hint >}}
+
+## Configuring workflows for function packages
+
+Function workflow definitions differ based on the base language the
+function implementation uses. For example, a Python function requires
+a Python environment in the GitHub Action runner:
+
+```yaml
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Setup Hatch
+        run: pipx install hatch==1.7.0
+
+      - name: Lint
+        run: hatch run lint:check
+```
+
+While the template repository provides a working pipeline definition, users may
+choose to customize their environment with different tooling.
+
+Functions also require a runtime image of the core business logic to
+build and embed into the Function package. The default workflow definition
+builds for two platforms: `linux/amd64` and `linux/arm64`.
+
+```yaml
+      - name: Build Runtime
+        id: image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          platforms: linux/${{ matrix.arch }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          target: image
+          build-args:
+            PYTHON_VERSION=${{ env.PYTHON_VERSION }}
+          outputs: type=docker,dest=runtime-${{ matrix.arch }}.tar
+```
+
+## Configuring workflows for provider packages
+
+Providers, unlike Functions, use custom `make` targets in the [build submodule]
+for building and pushing Crossplane Provider packages.
+
+Configuring the workflow for a specific registry involves two steps:
+
+1. Updating the registry variables in the top-level `Makefile`.
+2. Referencing GitHub Actions Secrets for authorized credentials to the
+registry.
+
+### Configure target registry
+
+The provider template repository includes a top-level [`Makefile`](https://github.com/crossplane/upjet-provider-template/blob/main/Makefile).
+Edit the following variables to define the target registry:
+
+1. `XPKG_REG_ORGS` - a space-delimited list of target repositories.
+2. `XPKG_REG_ORGS_NO_PROMOTE` - for registries that don't use or infer
+channel tags, such as `xpkg.upbound.io`.
+
+For example, the following dual-pushes to `xpkg.upbound.io` as well as
+`index.docker.io`:
+
+```make
+XPKG_REG_ORGS ?= xpkg.upbound.io/crossplane-contrib index.docker.io/crossplanecontrib
+
+XPKG_REG_ORGS_NO_PROMOTE ?= xpkg.upbound.io/crossplane-contrib
+```
+
+## Reusable workflows
+
+The [crossplane-contrib/provider-workflows] repository provide reusable
+workflow definitions that are callable from a custom CI pipeline.
+
+For example, the following snippet references the callable workflow to
+build and push the `provider-kubernetes` package to both `ghcr.io` and
+`xpkg.upbound.io`:
```

Comment:
maybe add a quick mention of the quickstarts to tie this back together in the reader's mind?
```suggestion
For example, the following snippet references the callable workflow to
build and push the `provider-kubernetes` package to both `ghcr.io` and
`xpkg.upbound.io`, as shown in the provider quickstart above:
```